errors/context.go

```diff
@@ -0,0 +1,42 @@
+package errors
+
+type errorWithContext interface {
```

Comment:
Also, the prefix is redundant. I think we should go with `withContext`, and methods like `SetContext()` instead of `ErrorSetContext()`.